prov/efa/test/efa_unit_test_hmem.c

```diff
@@ -49,9 +49,109 @@ void test_efa_hmem_info_update_neuron(struct efa_resource **state)
         assert_false(efa_domain->hmem_info[FI_HMEM_NEURON].initialized);
         assert_false(efa_domain->hmem_info[FI_HMEM_NEURON].p2p_supported_by_device);
 }
+
+/**
+ * @brief Verify when p2p is disabled, we don't check p2p support with neuron_alloc.
+ * Just leave p2p_supported_by_device to false.
+ *
+ * @param[in]	state		struct efa_resource that is managed by the framework
+ */
+void test_efa_hmem_info_disable_p2p_neuron(struct efa_resource **state)
+{
+        int ret;
+        struct efa_resource *resource = *state;
+        struct efa_domain *efa_domain;
+        uint32_t efa_device_caps_orig;
+        bool neuron_initialized_orig;
+
+	resource->hints = efa_unit_test_alloc_hints(FI_EP_RDM);
+        assert_non_null(resource->hints);
+
+        ret = fi_getinfo(FI_VERSION(1, 14), NULL, NULL, 0ULL, resource->hints, &resource->info);
+        assert_int_equal(ret, 0);
+
+        ret = fi_fabric(resource->info->fabric_attr, &resource->fabric, NULL);
+        assert_int_equal(ret, 0);
+
+        neuron_initialized_orig = hmem_ops[FI_HMEM_NEURON].initialized;
+        hmem_ops[FI_HMEM_NEURON].initialized = true;
+        efa_device_caps_orig = g_device_list[0].device_caps;
+        g_device_list[0].device_caps |= EFADV_DEVICE_ATTR_CAPS_RDMA_READ;
+        /* neuron_alloc should not be called when p2p is disabled */
+        g_efa_unit_test_mocks.neuron_alloc = &efa_mock_neuron_alloc_return_mock;
+
```

Comment:
for safe, add a `expect_function_call(efa_mock_neuron_alloc_return_mock);` here. Same thing for `ofi_cudaMalloc` mock

Reply:
Why adding `expect_function_call` when we expect `efa_mock_neuron_alloc_return_mock` not being called?

Reply:
Sorry, I missed it. Then how do we verify it is not called?

Reply:
`efa_mock_neuron_alloc_return_mock` just returns `mock()`, which will fail when `neuron_alloc` is called

Reply:
Hmmm this seems odd - can you try if `expect_function_calls(efa_mock_neuron_alloc_return_mock, 0)` works here?

Reply:
cmocka does not support 0 for expect_function_calls 
 https://github.com/clibs/cmocka/blob/master/src/cmocka.c#L1111

Right now efa_mock_neuron_alloc_return_mock will print 
 ```
No entries for symbol efa_mock_neuron_alloc_return_mock.
prov/efa/test/efa_unit_test_mocks.c:313: error: Could not get value to mock function efa_mock_neuron_alloc_return_mock
There were no previously returned mock values for this test.
```
if neuron_alloc is called. 

Or I can do a `assert_true(false)` in efa_mock_neuron_alloc_return_mock instead.

Reply:
I am ok with just adding some comment to explain why do you do a mock and why it can make sure the function is not called